providers/oidc.go

```diff
@@ -181,9 +181,18 @@ func (p *OIDCProvider) redeemRefreshToken(ctx context.Context, s *sessions.Sessi
 func (p *OIDCProvider) CreateSessionFromToken(ctx context.Context, token string) (*sessions.SessionState, error) {
 	idToken, err := p.Verifier.Verify(ctx, token)
 	if err != nil {
-		return nil, err
+		for _, jwtBearerVerifier := range p.JWTBearerVerifiers {
```

Comment:
So this is all handled already in this session chain in our session handling middleware:

https://github.com/oauth2-proxy/oauth2-proxy/blob/b541805dc1a7ab0dceba9959a31e4c7a06c7c6cf/oauthproxy.go#L261

If the goal is to get `groups` extraction out of secondary issuer JWTs, this is where we want to investigate: https://github.com/oauth2-proxy/oauth2-proxy/blob/b541805dc1a7ab0dceba9959a31e4c7a06c7c6cf/pkg/apis/middleware/session.go#L20

But as I mentioned in the original issue thread -- there's serious issues conflating provider specific settings with the generic bearer JWT handler. Last time we had PRs that tried that, it resulted in some nasty bugs for non OIDC provider users that took us a while to figure out the root cause.

We are looking to deprecate `--extra-jwt-issuers` in the next major version and replace it with chainable "Bearer Token" providers, where we will call the `CreateSessionFromToken` function of all the providers in that list. And because it will be after we have structured configs, those chainable bearer providers will get the ability to use rich, full configurations

Reply:
If we want a bandaid until v8 -- I think we do barebones `groups` extraction from extra ID Tokens in the `middlewareapi.CreateTokenToSessionFunc`, and only support `groups` explicitly named as the claim with no substitutes, and no non-standard formats (`[]string` only support).

Reply:
Understood. As a user of oauth2-proxy, I find the current behavior quite confusing though, and I would expect --extra-jwt-issuers to apply to the main provider and not create a default provider, as is done now.

Reply:
That was where we ran into issues in the past, the PRs that mixed `OIDCProvider` functionality into the `--extra-jwt-issuers` broke that functionality for all the users using `--extra-jwt-issuers` with Google, Gitlab & Keycloak as their main interactive providers with supplemental OIDC bearer tokens allowed.

Hopefully the future work for chainable bearer providers and making more providers based on OIDCProvider as a base where applicable will fix up a lot of these areas.

Reply:
The other issue we will need to figure out - the `Authorize` method, while mostly consistent, is overridable by the providers. So that's another area we need to figure out for `--extra-jwt-issuers` setting a `session.Groups` for use in the `Authorize` method (beyond just setting `X-Forwarded-Groups`).

> Understood. As a user of oauth2-proxy, I find the current behavior quite confusing though, and I would expect --extra-jwt-issuers to apply to the main provider and not create a default provider, as is done now.

Oh! And I remember the major bug this caused - even when everything was the OIDCProvider. Users that had OIDCProvider as their main provider with an interactive IdP like Okta, OneLogin, etc (which had full, spec complaint claims) mixed with machine tokens (e.g. Kubernetes) in `--extra-jwt-issuers` which didn't have the same claims at all (Kubernetes is missing email & groups for instance) stopped working when main OIDCProvider settings were forced onto the `--extra-jwt-issuers`